znai-java/src/main/java/com/twosigma/znai/java/parser/JavaCodeVisitor.java

```diff
@@ -187,19 +187,21 @@ private String extractJavaDocDescription(JavadocComment javadocComment) {
         return description == null ? "" : extractJavaDocDescription(description);
     }
 
-    @SuppressWarnings("unchecked")
     private String extractJavaDocDescription(JavadocDescription description) {
         // TODO check if github java parser lib solved the problem with inlined tags UnsupportedOperation exception
         List<JavadocDescriptionElement> elements = getPrivateFieldValue(description,"elements");
-        return elements.stream().map(this::elementToText).collect(joining(" "));
+        return elements.stream()
+                .map(this::elementToText)
+                .filter(text -> !text.isEmpty())
+                .collect(joining(" "));
     }
 
     private String elementToText(JavadocDescriptionElement el) {
         if (el instanceof JavadocSnippet) {
```

Comment:
Implement elementToText at `JavadocDescriptionElement`, so you can only call `el.elementToText()` here.
Also only `JavadocSnippet` and `JavadocInlineTag` implemented `JavadocDescriptionElement`.
The last `return` is unreachable or it can be NPE.

Reply:
`JavadocDescriptionElement` is a library class that has some of the mentioned in todo bugs. Hence the workaround. Not sure I follow about NPE. It is there as a fallback in case more implementations are added later.

Reply:
Got it. I thought that `JavadocDescriptionElement` is one in znai code base, so I suggested to move the logic of `elementToText` into the interface.
Currently there are two implementations of `JavadocDescriptionElement` and the code covers both of them, so the remaining case to reach the end `return` is `null`.